tests/test_scripts.py

```diff
@@ -561,6 +567,100 @@ def test_workspace_digest(tmpdir, script_runner, algorithms, do_json):
+    ],
+)
+def test_patchset_download(datadir, script_runner, archive):
+    command = f'pyhf contrib download {archive} {datadir.join("likelihoods").strpath}'
+    ret = script_runner.run(*shlex.split(command))
+    assert ret.success
+
+    # Run with all optional flags
+    command = f'pyhf contrib download --verbose --force {archive} {datadir.join("likelihoods").strpath}'
+    ret = script_runner.run(*shlex.split(command))
+    assert ret.success
+
+    command = f'pyhf contrib download --verbose https://www.fail.org/record/resource/1234567 {datadir.join("likelihoods").strpath}'
+    ret = script_runner.run(*shlex.split(command))
+    assert not ret.success
+    assert (
+        "pyhf.exceptions.InvalidArchiveHost: www.fail.org is not an approved archive host"
+        in ret.stderr
+    )
+    command = f'pyhf contrib download --verbose --force https://www.fail.org/record/resource/1234567 {datadir.join("likelihoods").strpath}'
+    ret = script_runner.run(*shlex.split(command))
+    assert not ret.success
+    # TODO: https://github.com/scikit-hep/pyhf/issues/1075
+    # Python 3.6 has different return error than 3.7, 3.8
+    assert (
+        "ssl.CertificateError: hostname 'www.fail.org' doesn't match"
+        or "certificate verify failed: Hostname mismatch, certificate is not valid for 'www.fail.org'."
+        in ret.stderr
+    )
+
+
+def test_missing_contrib_extra(caplog):
+    with mock.patch.dict(sys.modules):
+        sys.modules["requests"] = None
+        if "pyhf.contrib.utils" in sys.modules:
+            reload(sys.modules["pyhf.contrib.utils"])
+        else:
+            import_module("pyhf.cli")
+
+    with caplog.at_level(logging.ERROR):
+        assert (
+            "import of requests halted; None in sys.modules"
+            + "\nInstallation of the contrib extra is required to use pyhf.contrib.utils.download"
+            + "\nPlease install with: python -m pip install pyhf[contrib]"
+            in caplog.text
+        )
+        caplog.clear()
+
+
+def test_missing_contrib_download(caplog):
+    with mock.patch.dict(sys.modules):
+        sys.modules["requests"] = None
+        if "pyhf.cli" in sys.modules:
+            reload(sys.modules["pyhf.cli"])
+        else:
+            import_module("pyhf.cli")
+
+        # Force environment for runner
+        for module in [
+            "pyhf.cli.contrib",
+            "pyhf.contrib",
+            "pyhf.contrib.utils",
+        ]:
+            if module in sys.modules:
+                del sys.modules[module]
+
+        from pyhf.cli.contrib import download
+
+        runner = CliRunner(mix_stderr=False)
+        result = runner.invoke(
+            download,
+            [
+                "--verbose",
+                "https://www.hepdata.net/record/resource/1408476?view=true",
+                "1Lbb-likelihoods",
+            ],
+        )
+        assert result.exit_code == 0
+
+        with caplog.at_level(logging.ERROR):
```

Comment:
this is a really strict test. Can we just check for important pieces of information? E.G. "pyhf[contrib]" and the "no attribute download..." part? I'm fine with it as it is, but the more strict we make it, the more we need to maintain it.

Reply:
I added this just to make sure that we're properly hiding `download` from users, but I can relax this.

Reply:
@kratsg Oh, sorry I misread that before I pushed. You want the `module 'pyhf.contrib.utils' has no attribute 'download'` in still, correct? If so, then what is the strict part? I guess I'm unclear on what you want to have tested.